src/modules/components/transaction/transactionDataListItem/transactionDataListItemStructure/transactionDataListItemStructure.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+    AvatarIcon,
+    DataList,
+    Heading,
+    IconType,
+    NumberFormat,
+    Spinner,
+    formatterUtils,
+    type AvatarIconVariant,
+} from '../../../../../core';
+import { QueryType, createBlockExplorerLink } from '../../../../utils/blockExplorerUtils/blockExplorerUtils';
+import { formatDate } from '../../../../utils/timestampUtils';
+import {
+    TransactionType,
+    TxStatusCode,
+    txHeadingStringList,
+    txIconTypeList,
+    txVariantList,
+    type ITransactionDataListItemProps,
+} from './transactionDataListItemStructure.api';
+
+export const TransactionDataListItemStructure: React.FC<ITransactionDataListItemProps> = (props) => {
+    const {
+        chainId,
+        tokenAddress,
+        tokenSymbol,
+        tokenValue,
+        usdEstimate,
+        txType,
+        txStatus = TxStatusCode.PENDING,
+        unixTimestamp,
+        txHash,
+        ...otherProps
+    } = props;
+
+    const getEffectiveStatus = () => {
+        const type = txType;
+        if (txStatus === TxStatusCode.FAILED) {
+            return {
+                icon: IconType.CLOSE,
+                variant: 'critical' as AvatarIconVariant,
+                heading: 'Failed transaction',
+            };
```

Comment:
Why not add the heading, icon, and variant to the `txHeadingStringList`,`txIconTypeList`, and `txVariantList` respectively? That would make it so that you don't need this `getEffectiveStatus` function.

Reply:
Because Failed is ostensibly a TransactionType printout as the result of the status, but semantically it is not a type of transaction.

Reply:
Aha, I see. Makes sense. I would revisit the function name though 🙈